app/components/RegisterSingleTree/MapMarking.js

```diff
@@ -18,160 +16,153 @@ import Config from 'react-native-config';
+  const camera = useRef(null);
+  const map = useRef(null);
+  const navigation = useNavigation();
 
-  async UNSAFE_componentWillMount() {
+  useEffect(() => {
+    // Do something
     if (IS_ANDROID) {
-      MapboxGL.setTelemetryEnabled(false);
-      await MapboxGL.requestAndroidLocationPermissions().then(() => {});
+      MapboxGL.requestAndroidLocationPermissions().then((permission) => {
+        if (permission) {
+          MapboxGL.setTelemetryEnabled(false);
+          updateCurrentPosition();
+        }
+      });
+    } else {
+      updateCurrentPosition();
     }
-  }
-
-  componentDidMount() {
-    const { inventoryID } = this.props;
+    console.log(inventoryState, 'inventoryState');
+    const { inventoryID } = inventoryState.inventoryID;
     getInventory({ inventoryID: inventoryID }).then((inventory) => {
-      this.setState({ inventory: inventory });
-      if (inventory.polygons.length !== 0) {
-        const { latitude, longitude } = inventory.polygons[0].coordinates[0];
-        // this.onUpdateUserLocation([longitude, latitude]);
-        this.setState({ markedCoords: [longitude, latitude] });
-      }
+      setInventory(inventory);
     });
-  }
+  }, []);
 
-  renderFakeMarker = () => {
+  const renderFakeMarker = () => {
     return (
       <View style={styles.fakeMarkerCont}>
         <SvgXml xml={active_marker} style={styles.markerImage} />
-        {/* {this.state.loader ? (
-          <ActivityIndicator color={Colors.WHITE} style={styles.loader} />
-        ) : (
-          <Text style={styles.activeMarkerLocation}>{'A'}</Text>
-        )} */}
       </View>
     );
   };
 
-  onChangeRegionStart = () => this.setState({ loader: true });
+  const onChangeRegionStart = () => setLoader(true);
 
-  onChangeRegionComplete = async () => {
-    const center = await this._map.getCenter();
-    this.setState({ centerCoordinates: center, loader: false });
+  const onChangeRegionComplete = async () => {
+    const center = await map.current.getCenter();
+    setCenterCoordinates(center);
+    setLoader(false);
   };
 
-  onUpdateUserLocation = (location) => {
+  const onUpdateUserLocation = (location) => {
     if (!location) {
-      // alert('Unable to retrieve location')
       return;
     }
-    if (!this.state.isInitial) {
-      const currentCoords = [location.coords.longitude, location.coords.latitude];
-      // const currentCoords = location;
-      this.setState({ centerCoordinates: currentCoords, isInitial: true });
-      this._camera.setCamera({
-        centerCoordinate: currentCoords,
-        zoomLevel: 18,
-        animationDuration: 2000,
-      });
+    if (!isInitial) {
+      onPressMyLocationIcon(location);
     }
   };
-
-  addMarker = async () => {
-    let { centerCoordinates } = this.state;
+  const onPressMyLocationIcon = (position) => {
+    var recenterCoords;
+    if (position) {
+      recenterCoords = [position.coords.longitude, position.coords.latitude];
+    } else {
```

Comment:
When clicking on the icon to go to my position I get a crash: `TypeError: null is not an object (evaluating 'location.coords')` 

![Simulator Screen Shot - iPhone 11 - 2021-02-23 at 14 06 45](https://user-images.githubusercontent.com/1532418/108847936-a9649f00-75e0-11eb-874f-006bdd0d82f7.png)